### DISCUSSION TITLE
Looking for MeshGenerator that combines nodesets but leaves old nodesets intact

### DISCUSSION THREAD

**Discussion by jessecarterMOOSE** [idaholab/moose#25707] [Q&A General]
I have some nodesets/boundaries that I'd like to combine in multiple ways. Say I have 3 nodesets A, B, and C. I want:

* A + B = D
* A + C = E
* B + C = F

Seems like RenameBoundaryGenerator is close, but it destroys the old boundaries. I'm using it like:
```
old_boundary = 'A B'
new_boundary  = 'D D'
```
etc...

**Comment by GiudGiud** [comment]
Use ParsedGenerateSideset with the included_boundaries parameter to create new ones.

the combinatorial geometry can be `x > -1000000` to not matter

**Reply by jessecarterMOOSE** [reply]
Hmm, that creates sidesets, but what about nodesets? Also, the nodesets may not be located together - could be 4 corners of the mesh for example, or centers of mesh edges, so I can't think of a expression that works for the combinatorial geometry.

**Reply by GiudGiud** [reply]
kk seems like you ll need to make ParsedGenerateNodeset. It will be a massive copy paste with very few line changes
